controlplane/pkg/apis/local/connection/connection_helpers.go

```diff
@@ -55,7 +55,7 @@ func (c *Connection) SetContext(context *connectioncontext.ConnectionContext) {
 
 // UpdateContext checks and tries to set connection context
 func (c *Connection) UpdateContext(context *connectioncontext.ConnectionContext) error {
-	if err := context.MeetsRequirements(c.Context); err != nil {
+	if err := context.IpContext.MeetsRequirements(c.Context.IpContext); err != nil {
```

Comment:
I suppose we need to have MeetsRequirements() for top level context, and delegate to individuals?